### ISSUE TITLE
Add browse button to link field

### ISSUE THREAD

**Issue by matthew-d-jones** [ess-dmsc/nexus-constructor#391]
Link fields can be added by entering a text path, added in #330.
We can make life easier for users by adding some kind of "browse" button functionality. Perhaps bringing up modal with another silx treeview so a group or dataset can be selected?

**Comment by matthew-d-jones** [comment]
Blocked by #330

**Comment by matthew-d-jones** [comment]
Close as "won't do", can revive this idea later if user feedback suggests we need it.
